lib/Base/Organization/FilteredCollection.vala

```diff
@@ -0,0 +1,147 @@
+/* FilteredCollection.vala
+ *
+ * Copyright 2023 CodedOre <47981497+CodedOre@users.noreply.github.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ *
+ * FilteredCollection was adapted from code of Gtk.FilterListModel,
+ * created in 2015 by Benjamin Otte and licensed under the LGPL-2.1-or-later.
+ */
+
+using GLib;
+
+/**
+ * Extends Collection with the option to filter for specific items.
+ */
+public abstract class Backend.FilteredCollection<T> : Backend.Collection<T> {
+
+  /**
+   * Run at construction of an instance.
+   */
+  construct {
+    matches = new Gtk.Bitset.empty ();
+  }
```

Comment:
I wasn't aware of the GTK Bitset.

I'm assuming it's the right type of collection. It says "set" in the description, but it feels like we need an ordered list (because otherise `matches.get_nth()` wouldn't be guaranteed to get the item we're expecting). But it also says that it's most commonly used for complex selections, which is effectively what we're doing.